### PR TITLE
Reset peer and host IP after HTX runs

### PR DESCRIPTION
**Pull request by vaishnavibhat**
The patch addresses the following:
- Reset host IP with original IP
- Reset peer IP with original IP
- Add yaml parameters like peer_ips, netmask, hbond check.

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
#nohup python avocado-setup.py --run-suite host_io_network_fvt --input-file input.txt & 

#tail -f nohup.out 
02:37:53 INFO    : Check for environment
02:37:53 INFO    : Creating temporary mux dir
02:37:54 INFO    : 
02:37:54 INFO    : Running Host Tests Suite io_network_fvt_htx_nic_devices_HtxNicTest_test_start_htx_nic_devices
02:37:54 INFO    : Running: /usr/local/bin/avocado run --max-parallel-tasks=1 /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_start$ -m /tmp/mux/io_network_fvt_htx_nic_devices_HtxNicTest_test_start_htx_nic_devices.yaml --force-job-id e37f81136b4056bcc6432a38cf7cdadd4d22467a  --job-results-dir /root/avocado-fvt-wrapper/results
JOB ID     : e37f81136b4056bcc6432a38cf7cdadd4d22467a
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.37-e37f811/job.log
 (1/1) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_start;run-f3c6: STARTED
 (1/1) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_start;run-f3c6:  PASS (245.90 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.37-e37f811/results.html
JOB TIME   : 261.38 s
02:42:18 INFO    : 
02:42:18 INFO    : 
02:42:18 INFO    : Running Host Tests Suite io_network_fvt_htx_nic_devices_HtxNicTest_test_check_htx_nic_devices
02:42:18 INFO    : Running: /usr/local/bin/avocado run --max-parallel-tasks=1 /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_check$ -m /tmp/mux/io_network_fvt_htx_nic_devices_HtxNicTest_test_check_htx_nic_devices.yaml --force-job-id 0f636fed025ffb23f4515d4446cdccd45b0e17cb  --job-results-dir /root/avocado-fvt-wrapper/results
JOB ID     : 0f636fed025ffb23f4515d4446cdccd45b0e17cb
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.42-0f636fe/job.log
 (1/1) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_check;run-f3c6: STARTED
 (1/1) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_check;run-f3c6:  PASS (122.93 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.42-0f636fe/results.html
JOB TIME   : 138.88 s
02:44:40 INFO    : 
02:44:40 INFO    : 
02:44:40 INFO    : Running Host Tests Suite io_network_fvt_htx_nic_devices_HtxNicTest_test_stop_htx_nic_devices
02:44:40 INFO    : Running: /usr/local/bin/avocado run --max-parallel-tasks=1 /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_stop$ -m /tmp/mux/io_network_fvt_htx_nic_devices_HtxNicTest_test_stop_htx_nic_devices.yaml --force-job-id d1f35e216c73b6b341ae2b46536060c519fea9ca  --job-results-dir /root/avocado-fvt-wrapper/results
JOB ID     : d1f35e216c73b6b341ae2b46536060c519fea9ca
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.44-d1f35e2/job.log
 (1/1) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_stop;run-f3c6: STARTED
 (1/1) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/htx_nic_devices.py:HtxNicTest.test_stop;run-f3c6:  PASS (7.33 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.44-d1f35e2/results.html
JOB TIME   : 23.13 s
02:45:07 INFO    : 
02:45:07 INFO    : 
02:45:07 INFO    : Running Host Tests Suite io_network_fvt_network_test_network_test_virt_net
02:45:07 INFO    : Running: /usr/local/bin/avocado run --max-parallel-tasks=1 /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py -m /tmp/mux/io_network_fvt_network_test_network_test_virt_net.yaml --force-job-id 2cd5c7ad0c14787be688a3542416e4d126f15ea7  --job-results-dir /root/avocado-fvt-wrapper/results
JOB ID     : 2cd5c7ad0c14787be688a3542416e4d126f15ea7
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.45-2cd5c7a/job.log
 (01/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gro;run-mtu-1500-79e7: STARTED
 (01/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gro;run-mtu-1500-79e7:  PASS (6.04 s)
 (02/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gro;run-mtu-9000-0392: STARTED
 (02/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gro;run-mtu-9000-0392:  PASS (15.13 s)
 (03/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gso;run-mtu-1500-79e7: STARTED
 (03/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gso;run-mtu-1500-79e7:  PASS (5.67 s)
 (04/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gso;run-mtu-9000-0392: STARTED
 (04/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_gso;run-mtu-9000-0392:  PASS (15.14 s)
 (05/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_lro;run-mtu-1500-79e7: STARTED
 (05/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_lro;run-mtu-1500-79e7:  CANCEL: Unsupported on vNIC (5.61 s)
 (06/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_lro;run-mtu-9000-0392: STARTED
 (06/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_lro;run-mtu-9000-0392:  CANCEL: Unsupported on vNIC (14.89 s)
 (07/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_tso;run-mtu-1500-79e7: STARTED
 (07/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_tso;run-mtu-1500-79e7:  PASS (5.66 s)
 (08/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_tso;run-mtu-9000-0392: STARTED
 (08/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_tso;run-mtu-9000-0392:  PASS (15.19 s)
 (09/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ping;run-mtu-1500-79e7: STARTED
 (09/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ping;run-mtu-1500-79e7:  PASS (14.99 s)
 (10/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ping;run-mtu-9000-0392: STARTED
 (10/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ping;run-mtu-9000-0392:  PASS (24.36 s)
 (11/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_floodping;run-mtu-1500-79e7: STARTED
 (11/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_floodping;run-mtu-1500-79e7:  PASS (5.68 s)
 (12/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_floodping;run-mtu-9000-0392: STARTED
 (12/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_floodping;run-mtu-9000-0392:  PASS (14.86 s)
 (13/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ipv6_ping;run-mtu-1500-79e7: STARTED
 (13/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ipv6_ping;run-mtu-1500-79e7:  CANCEL: Test failing while getting IPV6 address for peer interface (5.64 s)
 (14/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ipv6_ping;run-mtu-9000-0392: STARTED
 (14/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ipv6_ping;run-mtu-9000-0392:  CANCEL: Test failing while getting IPV6 address for peer interface (15.02 s)
 (15/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ssh;run-mtu-1500-79e7: STARTED
 (15/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ssh;run-mtu-1500-79e7:  PASS (5.69 s)
 (16/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ssh;run-mtu-9000-0392: STARTED
 (16/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_ssh;run-mtu-9000-0392:  PASS (15.15 s)
 (17/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_scp;run-mtu-1500-79e7: STARTED
 (17/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_scp;run-mtu-1500-79e7:  PASS (15.92 s)
 (18/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_scp;run-mtu-9000-0392: STARTED
 (18/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_scp;run-mtu-9000-0392:  PASS (24.87 s)
 (19/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_jumbo_frame;run-mtu-1500-79e7: STARTED
 (19/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_jumbo_frame;run-mtu-1500-79e7:  PASS (8.80 s)
 (20/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_jumbo_frame;run-mtu-9000-0392: STARTED
 (20/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_jumbo_frame;run-mtu-9000-0392:  PASS (18.16 s)
 (21/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_statistics;run-mtu-1500-79e7: STARTED
 (21/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_statistics;run-mtu-1500-79e7:  PASS (5.64 s)
 (22/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_statistics;run-mtu-9000-0392: STARTED
 (22/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_statistics;run-mtu-9000-0392:  PASS (15.04 s)
 (23/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_promisc;run-mtu-1500-79e7: STARTED
 (23/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_promisc;run-mtu-1500-79e7:  PASS (9.75 s)
 (24/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_promisc;run-mtu-9000-0392: STARTED
 (24/24) /root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/network_test.py:NetworkTest.test_promisc;run-mtu-9000-0392:  PASS (18.60 s)
RESULTS    : PASS 20 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-10-08T02.45-2cd5c7a/results.html
JOB TIME   : 366.74 s
02:51:16 INFO    : 
02:51:16 INFO    : Summary of test results can be found below:
TestSuite                                                                                 TestRun    Summary
 
host_io_network_fvt_htx_nic_devices_HtxNicTest_test_start_htx_nic_devices                 Run        Successfully executed
/root/avocado-fvt-wrapper/results/job-2024-10-08T02.37-e37f811/job.log
| PASS 1 || CANCEL 0 || ERRORS 0 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
 
host_io_network_fvt_htx_nic_devices_HtxNicTest_test_check_htx_nic_devices                 Run        Successfully executed
/root/avocado-fvt-wrapper/results/job-2024-10-08T02.42-0f636fe/job.log
| PASS 1 || CANCEL 0 || ERRORS 0 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
 
host_io_network_fvt_htx_nic_devices_HtxNicTest_test_stop_htx_nic_devices                  Run        Successfully executed
/root/avocado-fvt-wrapper/results/job-2024-10-08T02.44-d1f35e2/job.log
| PASS 1 || CANCEL 0 || ERRORS 0 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
 
host_io_network_fvt_network_test_network_test_virt_net                                    Run        Successfully executed
/root/avocado-fvt-wrapper/results/job-2024-10-08T02.45-2cd5c7a/job.log
| PASS 20 || CANCEL 4 || ERRORS 0 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
02:51:16 INFO    : Removing temporary mux dir
^C[1]+  Done                    nohup python avocado-setup.py --run-suite host_io_network_fvt --input-file input.txt

**Comment by abdhaleegit** [comment]
@vaishnavibhat is this PR related to this https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2864

File: io/net/htx_nic_devices.py [hunk 1]
```diff
@@ -68,22 +68,26 @@ def setUp(self):
...
+                    networkinterface.save(ipaddr, self.netmask)
+            for interface in self.peer_intfs:
+                peer_networkinterface = NetworkInterface(interface, self.remotehost)
+            for peerip in self.peer_ipaddrs:
+                try:
+                    peer_networkinterface.add_ipaddr(peerip, self.netmask)
+                    peer_networkinterface.save(peerip, self.netmask)
+                except Exception:
+                    peer_networkinterface.save(peerip, self.netmask)
+        self.ip_addr_remove()
```
Comment (abdhaleegit) [inline io/net/htx_nic_devices.py:90]: if you want to flush the IP, why to configure in the first place ?... incase to take a backup of ip configuration.. than every test case does configure network before start...  so just flush and delete the ips and than let htx configure its own ips.... and at the end of htx again.. flush out the thxips  ? am I missing something here ?

File: io/net/htx_nic_devices.py [hunk 2]
```diff
@@ -375,12 +396,38 @@ def ip_restore_peer(self):
...
+        # Flush out the ip addresses on peer
+        for peer_interface in self.peer_intfs:
+            cmd = 'ip addr flush dev %s' % peer_interface
+            self.session.cmd(cmd)
+            cmd = 'ip link set dev %s up' % peer_interface
+            self.session.cmd(cmd)
+
     def htx_cleanup(self):
         self.shutdown_htx_daemon()
+        self.ip_addr_remove()
```
Comment (abdhaleegit) [inline io/net/htx_nic_devices.py:430]: remove this as ip_restore already flushes ips

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
Is network configuration needed here ? if htx can take care of ip configurations .. let it .. only flush before and after test ?

Also how will this impact ... htx_start.. htx_check and htx_stop calls in cfg files for scenario cfg ?

**Pull request closed without merging**